src/VisualStudio/Core/Def/Implementation/FindReferences/FindUsagesValueUsageInfoColumnDefinition.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.FindUsages
+{
+    /// <summary>
+    /// Custom column to display the reference kind/usage info for the Find All References window.
+    /// </summary>
+    [Export(typeof(ITableColumnDefinition))]
+    [Name(ColumnName)]
+    internal sealed class FindUsagesValueUsageInfoColumnDefinition : AbstractFindUsagesCustomColumnDefinition
+    {
+        public const string ColumnName = nameof(ValueUsageInfo);
+
+        // Allow filtering of the column by each ValueUsageInfo kind.
+        private static readonly ImmutableArray<string> s_defaultFilters = Enum.GetValues(typeof(ValueUsageInfo))
+                                                                .Cast<ValueUsageInfo>()
+                                                                .Where(value => value != ValueUsageInfo.None && (value & (value - 1)) == 0) // Single bit set
+                                                                .Select(v => v.ToString())
+                                                                .ToImmutableArray();
```

Comment:
it's unclear to me what the localization concerns are here.  Are these filters shown to the user?  If so, we likely cna't use v.toString here and we need to actually provide appropriate localized values depending on the enum value.

Reply:
Yes, the values here are directly from the [ValueUsageInfo enum](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Extensions/ValueUsageInfo.cs,8). We either need to define corresponding string constants or localizable resource strings here to display in FAR column.

Reply:
gotcha.  as a WIP this is fine then :)

Reply:
This is pending design issue, which will be addressed once we finalize the design: allowable values in this column AND localizable or not.

---
In reply to: [220755788](https://github.com/dotnet/roslyn/pull/30155#discussion_r220755788) [](ancestors = 220755788)